app/models/cloud_service.rb

```diff
@@ -31,4 +31,11 @@ def validate_disable_scheduling
   def scheduling_enabled?
     !scheduling_disabled?
   end
+
+  def delete
```

Comment:
This is named too similarly to the `ActiveRecord.delete` class method.

Maybe try `CloudService#delete_service`.  It looks redundant (i.e., `service.delete_service`), but I think it's probably better than looking like the framework's class method.